test/device/main.c

```diff
@@ -143,6 +144,45 @@ int test_dev_shm() {
     return 0;
 }
 
+int test_dev_fd() {
+    char *file_path = "/root/hello_world";
+    char *greetings = "hello";
+    int fd = open(file_path, O_RDWR | O_CREAT | O_TRUNC, 00666);
+    if (fd < 0) {
+        THROW_ERROR("failed to open a file to write");
+    }
+
+    // Generate dev_fd path
+    char dev_fd_path[20] = "/dev/fd/";
+    char *fd_str = calloc(1, 4); // 4 bytes would be enough here.
+    if (fd_str == NULL) {
+        THROW_ERROR("calloc failed");
+    }
+    asprintf(&fd_str, "%d", fd);
+    strcat(dev_fd_path, fd_str);
+    int dev_fd = open(dev_fd_path, O_WRONLY, 0666);
+    if (dev_fd < 0) {
+        THROW_ERROR("failed to open %s", dev_fd_path);
+    }
+
+    int ret = write(dev_fd, greetings, strlen(greetings));
```

Comment:
use 'len' as the return value of write/read is more correct.

Reply:
Done.